src/strands_tools/bright_data.py

```diff
@@ -0,0 +1,486 @@
+            params={"dataset_id": dataset_id, "include_errors": True},
+            headers=self.headers,
+            json=[request_data],
+        )
+
+        trigger_data = trigger_response.json()
+        if not trigger_data.get("snapshot_id"):
+            raise Exception("No snapshot ID returned from trigger request")
+
+        snapshot_id = trigger_data["snapshot_id"]
+        if self.verbose:
+            print(f"[Bright Data] {source_type} triggered with snapshot ID: {snapshot_id}")
+
+        attempts = 0
+        max_attempts = timeout
+
+        while attempts < max_attempts:
+            try:
+                snapshot_response = requests.get(
+                    f"https://api.brightdata.com/datasets/v3/snapshot/{snapshot_id}",
+                    params={"format": "json"},
+                    headers=self.headers,
+                )
+
+                snapshot_data = snapshot_response.json()
+
+                if isinstance(snapshot_data, dict) and snapshot_data.get("status") == "running":
+                    if self.verbose:
+                        print(
+                            f"[Bright Data] Snapshot not ready, polling again (attempt {attempts + 1}/{max_attempts})"
+                        )
+                    attempts += 1
+                    time.sleep(polling_interval)
+                    continue
+
+                if self.verbose:
+                    print(f"[Bright Data] Data received after {attempts + 1} attempts")
+
+                return snapshot_data
+
+            except Exception as e:
+                if self.verbose:
+                    print(f"[Bright Data] Polling error: {e!s}")
+                attempts += 1
+                time.sleep(polling_interval)
+
+        raise TimeoutError(f"Timeout after {max_attempts} seconds waiting for {source_type} data")
+
+
+@tool
+def bright_data(
+    action: str,
+    url: Optional[str] = None,
+    output_path: Optional[str] = None,
+    zone: Optional[str] = None,
+    query: Optional[str] = None,
+    engine: str = "google",
+    language: Optional[str] = None,
+    country_code: Optional[str] = None,
+    search_type: Optional[str] = None,
+    start: Optional[int] = None,
+    num_results: int = 10,
+    location: Optional[str] = None,
+    device: Optional[str] = None,
+    return_json: bool = False,
+    source_type: Optional[str] = None,
+    num_of_reviews: Optional[int] = None,
+    timeout: int = 600,
+    polling_interval: int = 1,
+) -> str:
+    """
+    Web scraping and data extraction tool powered by Bright Data.
+
+    This tool provides a comprehensive interface for web scraping and data extraction using
+    Bright Data, including scraping web pages as markdown, taking screenshots, performing
+    search queries, and extracting structured data from various websites.
+
+    Args:
+    action: The action to perform (scrape_as_markdown, get_screenshot, search_engine, web_data_feed)
```

Comment:
During testing noticed zone names needed agent to figure which zones does exists, added in docstrings here

```suggestion
    action: The action to perform (scrape_as_markdown, get_screenshot, search_engine, web_data_feed, get_active_zones)
```